src/NLog/Targets/ZipArchiveFileCompressor.cs

```diff
@@ -50,17 +50,24 @@ internal class ZipArchiveFileCompressor : IFileCompressor
         /// </summary>
         public void CompressFile(string fileName, string archiveFileName)
         {
+            string entryName = IsEntryNameInferredFromArchiveFileName ? (Path.GetFileNameWithoutExtension(archiveFileName) + Path.GetExtension(fileName)) : fileName;
```

Comment:
Think it should be (Missing `Path.GetFileName`):
`string entryName = IsEntryNameInferredFromArchiveFileName ? (Path.GetFileNameWithoutExtension(archiveFileName) + Path.GetExtension(fileName)) : Path.GetFileName(fileName);`

Reply:
Of course - thank you